### DISCUSSION TITLE
How to find and assign SKU number to sku variable in iPXE?

### DISCUSSION THREAD

**Discussion by n010gic** [ipxe/ipxe#1250] [General]
Hello there,

I'm fighting with iPXE and i'm loosing.
Can you help me resolve the issue below:

Locations of iPXE server and files:
/pxe-boot
/pxe-boot/skucheck.sh
/pxe-boot/images/1234567#ABA.wim (and more the same format)
/pxe-boot/ipxe/BCD
/pxe-boot/ipxe/boot.sdi

access to folders and files:
sudo chmod a+x /pxe-boot/
sudo chmod a+x /pxe-boot/skucheck.sh

iPXE file:

#!ipxe
dhcp
cpuid --ext 29
kernel wimboot
initrd /ipxe/BCD
initrd /ipxe/boot.sdi
chain skucheck.sh
initrd /images/${sku}.wim
boot

skucheck.sh file:

#!/bin/bash
sku=$(dmidecode -s system-sku-number)
export sku

Output on laptop:

![image](https://github.com/ipxe/ipxe/assets/174742017/e6e0ab9d-7b7b-46bd-9576-52c1f9c89b9f)

**Comment by NiKiZe** [comment]
You are chaining in to skucheck.sh?
iPXE can not run bash files.

Get in to the iPXE shell, run the `config` command, it will show you a menu with some available settings, including smbios data, which might be what you want.

Another problem you will see .. have you tried loading the URL for the wimnfile in browser? you will probably get a 404, unless you urlencode the filename.

I don't really think it makes sense to have different wim files per SKU, instead deal with this once winpe has booted.
You might be able to find inspiration from https://github.com/NiKiZe/wimboot-install/tree/main/win10

**Reply by n010gic** [reply]
Thanks NiKiZe

I checked config. There is no SKU (only model, serial...), but i need SKU.
All ${sku}.wim files are images for each laptop.
I can make it to load WinPE and from there go and download right image, but i want to skip Winpe.
if i use serial number instead of SKU, the script below works perfectly.

#!ipxe
dhcp
cpuid --ext 29
kernel wimboot
initrd /ipxe/BCD
initrd /ipxe/boot.sdi
initrd /images/${serial}.wim
boot

**Reply by NiKiZe** [reply]
You can only boot winpe, not full images.
For normal images you should be able to skip BCD and boot.sdi.

You could check https://lists.ipxe.org/pipermail/ipxe-devel/2014-June/003562.html the SKU value should be there somewhere.
